### DISCUSSION TITLE
gdk-pixbuf-query-loaders not found build error during demo-image-sato build, kirkstone branch

### DISCUSSION THREAD

**Discussion by hutcruchi** [OE4T/meta-tegra#1328] [Q&A] [answered]
Hi,
I followed the instructions for "quick setup" in tegra-demo-distro README.

My build fails with the following error:

```
NOTE: > Executing update_gtk_icon_cache intercept ...
NOTE: Exit code 127. Output:
/tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/intercept_scripts-6b84778c215deffb57ceca469f1155d305f66de1f30a78d1bf9b82a20a0f2ff8/update_gtk_icon_cache: 10: /tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/recipe-sysroot-native//gdk-pixbuf-2.0/gdk-pixbuf-query-loaders: not found
```

1. I installed the libgdk-pixbuf2.0-dev package
2. Since its installed in /usr/lib I executed:
`sudo ln -s /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/gdk-pixbuf-query-loaders /usr/bin/gdk-pixbuf-query-loaders`

I still have the same error.

**Comment by hutcruchi** [comment]
The problem was that for some reason the following directory was missing:
`/tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/recipe-sysroot-native/gdk-pixbuf-2.0`
it looked for gdk-pixbuf-query-loaders in that folder so I solved it by:
`sudo ln -s /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/gdk-pixbuf-query-loaders /tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/recipe-sysroot-native/gdk-pixbuf-2.0/gdk-pixbuf-query-loaders`

now I have a different error:
```
NOTE: > Executing update_gtk_icon_cache intercept ...
NOTE: Failed to create file “/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders.cache.QGQD91”: Permission denied

NOTE: > Executing update_pixbuf_cache intercept ...
NOTE: Exit code 2. Output:
/tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/intercept_scripts-6b84778c215deffb57ceca469f1155d305f66de1f30a78d1bf9b82a20a0f2ff8/update_pixbuf_cache: 11: cannot create /tegra_demo/build/tmp/work/jetson_orin_nano_devkit-oe4t-linux/tegra-espimage/1.0-r0/rootfs/usr/lib/gdk-pixbuf-2.0/2.10.0/loaders/../loaders.cache: Directory nonexistent
```

I keep fixing one error at a time but maybe I'm missing something here. I followed the instructions in the README and nothing mentioned regarding these issues. Maybe I'm doing something wrong

**Reply by madisongh** [reply]
See my comment [over here](https://github.com/OE4T/meta-tegra/discussions/1325#discussioncomment-6739930).  This really sounds like a filesystem problem on your build host.  Checking and repairing it would be a good thing to try.

**Reply by hutcruchi** [reply, marked as answer]
formatting my ubuntu host solved the problem. here:
https://github.com/OE4T/meta-tegra/discussions/1325
